overlays/ghc-packages.nix

```diff
@@ -29,13 +29,60 @@ let
       cabal-to-nix *.cabal > $out
     '';
   };
-  importCabal = name: src:
+
+  # Combines multiple derivations into one to make them
+  # easier to materialize.
+  combineFiles = name: ext: files: self.stdenv.mkDerivation {
+    inherit name;
+    phases = [ "buildPhase" ];
+
+    buildPhase = ''
+      mkdir $out
+    '' + self.lib.concatStrings (self.lib.mapAttrsToList (name: file: ''
+      cp ${file} $out/${name}${ext}
+    '') files);
+  };
+
+  # Convert the calculate the sdist of a single boot
+  # and the output of cabal-to-nix
+  cabalToSdistAndNix = ghcName: pkgName: src:
     # build the source dist
-    let sdist = callCabalSdist name src;
+    let sdist = callCabalSdist "${ghcName}-${pkgName}" src;
     # and generate the nix expression corresponding to the source dist
-    # but fixing the src to the sdist as well.
-    in args: (import (callCabal2Nix sdist) args) // { src = sdist; };
+    in {
+      inherit sdist;
+      nix = callCabal2Nix "${ghcName}-${pkgName}" sdist;
+    };
+
+  # Combine the all the boot package nix files for a giving ghc
+  # into a single derivation and materialize it.
+  combineAndMaterialize = ghcName: bootPackages:
+      let
+        # Limiting this to "ghc865" as the not all the boot packages can
+        # for ghc 8.8 and above can be processed yet by `nix-tools`
+        # (we need to upgrade `nix-tools` to Cabal 3 for them to work)
```

Comment:
:grimacing:

Reply:
That comment made it sound like materialization was not used at all for ghc 8.8, but it is we just can't materialize the bits that don't work yet.  I have updated it.